### PR TITLE
new sidelining exper script

### PR DESCRIPTION
**Pull request by NishanthJKumar**

### CODE REVIEW

**Comment by ronuchit** [comment]
note: will have to change this in light of the changes i'm making rn

File: scripts/supercloud/run_predicators_sidelining_experiments.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+FILE="scripts/supercloud/submit_supercloud_job.py"
+# Note: this script is meant to be run first, to completion, with
+# RUN_LOAD_EXPERIMENTS=false, then rerun with RUN_LOAD_EXPERIMENTS=true.
+RUN_LOAD_EXPERIMENTS=false
+NUM_TRAIN_TASKS="50"
+ALL_ENVS=(
+    "repeated_nextto_painting"
+    "screws"
+    "repeated_nextto_single_option"
+    "painting"
```
Comment (tomsilver) [inline scripts/supercloud/run_predicators_sidelining_experiments.sh:14]: probs worth running this one locally on one seed to see what happens, if you haven't already

File: scripts/supercloud/run_predicators_sidelining_experiments.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            # Model-free GNN baseline.
+            python $FILE --experiment_id ${ENV}_gnn_modelfree_${NUM_TRAIN_TASKS}demo --load_experiment_id ${ENV}_gnn_shooting_${NUM_TRAIN_TASKS}demo --env $ENV --approach gnn_option_policy --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS --gnn_option_policy_solve_with_shooting False --load_approach --load_data
+
+        else
+            # Main backchaining approach.
+            python $FILE --experiment_id ${ENV}_backchaining_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner backchaining --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+
+            # Cluster-and-intersect based methods for which harmlessness is guaranteed.
+            # Cluster and Intersect (RLDM) baseline.
+            python $FILE --experiment_id ${ENV}_cluster_and_intersect_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner cluster_and_intersect --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS --disable_harmlessness_check True
```
Comment (tomsilver) [inline scripts/supercloud/run_predicators_sidelining_experiments.sh:29]: if harmlessness is guaranteed, why disable the checks?

File: scripts/supercloud/run_predicators_sidelining_experiments.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# RUN_LOAD_EXPERIMENTS=false, then rerun with RUN_LOAD_EXPERIMENTS=true.
+RUN_LOAD_EXPERIMENTS=false
+NUM_TRAIN_TASKS="50"
+ALL_ENVS=(
+    "repeated_nextto_painting"
+    "screws"
+    "repeated_nextto_single_option"
+    "painting"
+)
+
```
Comment (tomsilver) [inline scripts/supercloud/run_predicators_sidelining_experiments.sh:16]: using the $COMMON_ARGS pattern from the other scripts makes it easier to see the diffs between the approaches

File: scripts/supercloud/run_predicators_sidelining_experiments.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            python $FILE --experiment_id ${ENV}_backchaining_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner backchaining --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+
+            # Cluster-and-intersect based methods for which harmlessness is guaranteed.
+            # Cluster and Intersect (RLDM) baseline.
+            python $FILE --experiment_id ${ENV}_cluster_and_intersect_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner cluster_and_intersect --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS --disable_harmlessness_check True
+            # LOFT baseline.
+            python $FILE --experiment_id ${ENV}_cluster_and_search_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner cluster_and_search --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS --disable_harmlessness_check True
+            # Prediction-error baseline.
+            python $FILE --experiment_id ${ENV}_pred_error_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --strips_learner cluster_and_intersect_sideline_prederror --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS --disable_harmlessness_check True
+
```
Comment (tomsilver) [inline scripts/supercloud/run_predicators_sidelining_experiments.sh:34]: nit: either remove this line break or add them between the other approaches too

**Review by tomsilver — COMMENTED** [review]
looking good!

File: scripts/supercloud/run_predicators_sidelining_experiments.sh (hunk 2 above)
Comment (ronuchit) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:29]: i wrote that actually -- it's super slow...

Comment (ronuchit) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:29]: (if ops have high arity)

File: scripts/supercloud/run_predicators_sidelining_experiments.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+START_SEED=456
+NUM_SEEDS=10
+FILE="scripts/supercloud/submit_supercloud_job.py"
+# Note: this script is meant to be run first, to completion, with
+# RUN_LOAD_EXPERIMENTS=false, then rerun with RUN_LOAD_EXPERIMENTS=true.
+RUN_LOAD_EXPERIMENTS=false
+NUM_TRAIN_TASKS="50"
```
Comment (ronuchit) [inline scripts/supercloud/run_predicators_sidelining_experiments.sh:9]: nit: remove the quotes (doesn't affect anything)

**Review by ronuchit — CHANGES_REQUESTED** [review]
(blocking this so we don't accidentally merge; let's rebase off #921 before launching)

**Comment by ronuchit** [comment]
TODO(rohan): remove `run_strips_learner_experiments.sh` and incorporate its comments into this new file

File: scripts/supercloud/run_predicators_sidelining_experiments.sh (hunk 2 above)
Comment (NishanthJKumar) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:29]: yup - exactly what Rohan said. High arity ops tend to happen in most of our interesting domains

File: scripts/supercloud/run_predicators_sidelining_experiments.sh (hunk 1 above)
Comment (NishanthJKumar) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:14]: ran locally and it solved 50/50, BUT it ended up learning sidelined operators for placing (which is definitely unexpected). Probably something we should look into!

Comment (ronuchit) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:14]: can you paste the operators?

Comment (ronuchit) [reply scripts/supercloud/run_predicators_sidelining_experiments.sh:12]: (they're on slack)

**Review by ronuchit — APPROVED** [review]

**Pull request merged**